[CI] MixedClusterClientYamlTestSuiteIT test {p0=mtermvectors/10_basic/Tests catching other exceptions per item} failing

**Build Scans:**
- [elasticsearch-intake #38880 / 9.3.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mifnkcebpnz7w)
- [elasticsearch-intake #38864 / 9.2.9_bwc-snapshots](https://gradle-enterprise.elastic.co/s/3l54x46xrlqkc)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.3.4#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}" -Dtests.seed=5617EDA12D159C42 -Dtests.bwc=true -Dtests.locale=es-PR -Dtests.timezone=Europe/Vilnius -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dmtermvectors%2F10_basic%2FTests%20catching%20other%20exceptions%20per%20item%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 2 errors:
  java.lang.AssertionError(field [docs.0.term_vectors.text.terms.brown.term_freq] is null)
  java.lang.AssertionError(field [docs.0.term_vectors.text.terms.brown.ttf] is null)
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
- [main] 1 consecutive failures in step 9.3.4_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test test {p0=mtermvectors/10_basic/Tests catching other exceptions per item} (0.2% fail rate in 1000 executions)
- [main] 2 failures in pipeline elasticsearch-intake (2.4% fail rate in 84 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

when running this locally, it fails with different exceptions:
```
» [2026-05-11T13:50:41,521][WARN ][o.e.c.s.MasterService    ] [v9.3.4-1] Failing [elected-as-master ([3] nodes joined in term 6)[_FINISH_ELECTION_, {v9.3.4-2}{Yzo3CC4jQTyrhTxMufJ24g}{g6zkw-q9TyeWy0HaTUEVxA}{v9.3.4-2}{127.0.0.1}{127.0.0.1:53204}{cdfhilmrstw}{9.3.4}{8000099-9060001} completing election, {v9.3.4-1}{BPsgxstmQsqBDEC1AHCbjA}{nt7z3xZHQLmFR2sZLUvJrQ}{v9.3.4-1}{127.0.0.1}{127.0.0.1:53660}{cdfhilmrstw}{9.5.0}{8000099-9098000} completing election, {v9.3.4-3}{8rKXh8wiSAi4Gz2VnAxqxA}{5BhxLcgTQU2aXck2m4s4Kg}{v9.3.4-3}{127.0.0.1}{127.0.0.1:53210}{cdfhilmrstw}{9.3.4}{8000099-9060001} completing election]]: failed to commit cluster state version [27] org.elasticsearch.cluster.coordination.FailedToCommitClusterStateException: publication of cluster state version [27] in term [6] failed [committed=false]
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Coordinator$CoordinatorPublication$4.onFailure(Coordinator.java:2193)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener$FailureResult.complete(SubscribableListener.java:448)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:351)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:243)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Coordinator$CoordinatorPublication.onCompletion(Coordinator.java:2103)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Publication.onPossibleCompletion(Publication.java:114)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Publication.cancel(Publication.java:78)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Coordinator.lambda$cancelActivePublication$27(Coordinator.java:1820)
»       at java.base/java.util.Optional.ifPresent(Optional.java:182)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Coordinator.cancelActivePublication(Coordinator.java:1820)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Coordinator.becomeCandidate(Coordinator.java:992)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Coordinator.joinLeaderInTerm(Coordinator.java:675)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.JoinHelper.lambda$new$2(JoinHelper.java:150)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:86)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:319)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:331)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
»       at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
»       at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
»       at java.base/java.lang.Thread.run(Thread.java:1516)
»  Caused by: org.elasticsearch.ElasticsearchException: publication cancelled before committing: become candidate: joinLeaderInTerm
»       at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.cluster.coordination.Publication.cancel(Publication.java:75)
»       ... 14 more
````

9.2.9 has been pulled off the bwc branches, hence "it shouldn't fail anymore" 🙃

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -407,9 +407,6 @@ tests:
 - class: org.elasticsearch.search.aggregations.metrics.LargeTopHitsIT
   method: test500Queries
   issue: https://github.com/elastic/elasticsearch/issues/148056
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
-  issue: https://github.com/elastic/elasticsearch/issues/148096
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testStressWithRelocationFailures
   issue: https://github.com/elastic/elasticsearch/issues/148129

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mtermvectors/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mtermvectors/10_basic.yml
@@ -19,6 +19,14 @@ setup:
   - do:
       indices.refresh: {}
 
+  # In mixed-cluster BWC tests, the rebalancer may relocate shards after node
+  # upgrades. A request hitting a recovering shard copy returns empty term
+  # vectors. Wait for no relocating shards before proceeding.
+  - do:
+      cluster.health:
+        index: testidx
+        wait_for_no_relocating_shards: true
+
 ---
 "Basic tests for multi termvector get":
 

